scripts/albs-952_fill_errata_module.py

```diff
@@ -0,0 +1,30 @@
+import asyncio
+import os
+import re
+import sys
+from contextlib import asynccontextmanager
+
+from sqlalchemy.future import select
+sys.path.append(os.path.dirname(os.path.dirname(__file__)))
+
+from alws import models
+from alws.dependencies import get_db
+
+async def main():
+    module_regex = re.compile('Module ([\d\w\-\_]+:[\d\.\w]+) is enabled')
+
+    updated_records = []
+    async with asynccontextmanager(get_db)() as db, db.begin():
+        errata_records = (await db.execute(
+            select(models.ErrataRecord))).scalars().all()
+        for record in errata_records:
+            match = module_regex.findall(str(record.original_criteria))
+            if match:
+                record.module = match[0]
+                updated_records.append(record)
+        db.add_all(updated_records)
+        await db.commit()
```

Comment:
should we update modules in pulp too or we planning to fix them manually?

Reply:
Note that the script is meant to update the new column `module` of every errata_record in our db. 
Regarding pulp, as far as I can tell, "should" be fine, look, i.e.:
```
pulp=> select name, module from rpm_updatecollection where update_record_id in (select content_ptr_id from rpm_updaterecord where id='ALSA-2022:7822');
                        name                         |                                                          module                                                          
-----------------------------------------------------+--------------------------------------------------------------------------------------------------------------------------
 almalinux-8-for-i686-appstream-rpms__8_7_default    | {"arch": "i686", "name": "container-tools", "stream": "rhel8", "context": "20125149", "version": 8070020221108190354}
 almalinux-8-for-x86_64-appstream-rpms__8_7_default  | {"arch": "x86_64", "name": "container-tools", "stream": "rhel8", "context": "20125149", "version": 8070020221108190354}
 almalinux-8-for-aarch64-appstream-rpms__8_7_default | {"arch": "aarch64", "name": "container-tools", "stream": "rhel8", "context": "20125149", "version": 8070020221108190354}
 almalinux-8-for-ppc64le-appstream-rpms__8_7_default | {"arch": "ppc64le", "name": "container-tools", "stream": "rhel8", "context": "20125149", "version": 8070020221108190354}
 almalinux-8-for-s390x-appstream-rpms__8_7_default   | {"arch": "s390x", "name": "container-tools", "stream": "rhel8", "context": "20125149", "version": 8070020221108190354}
(5 rows)

```
It would be interesting to run this query in production though.
So yes, let's pair up tomorrow and do some double checking here and there

Reply:
For the record, we don't need to fix the modules but the affected updateinfo.xml files. We do know that our pulp modules aren't reliable (as they miss artifacts) and that we should honor the modules.yaml files that are in production.